libs/utils/src/lib/docs-versions.tsx

```diff
@@ -2,31 +2,46 @@
  * ********************************************
  * ** List of versions of the documentation  **
  * ********************************************
- *
- * For each version, you must specify 2 things:
- * Label - We will use this for the sidebar.
- * ID - We will use it to create folders.
- * Branch, tag or commit - Used to fetch data.
- *
  **/
 
-export interface DocsVersion {
+/**
+ * Produces an object type that must contain exactly one key (in union K) of type V
+ * See: https://stackoverflow.com/a/60873215
+ */
+type ExactlyOneKeyWithType<K extends keyof any, V, KK extends keyof any = K> =
+  { [P in K]: { [Q in P]: V } &
+    { [Q in Exclude<KK, P>]?: never} extends infer O ?
+    { [Q in keyof O]: O[Q] } : never
+  }[K];
+
+type BaseDocsVersion = {
+  /** Used for visual display, e.g. the docs sidebar */
   label: string;
+  /** Used for routes */
   id: string;
-  branch?: string;
-  tag?: string;
-  commit?: string;
+  /** Used in calculations */
+  number: number;
+  isPreRelease?: boolean;
 }
 
+/** Used to fetch data. Only one type may be provided. */
+type DocsVersionReferenceKind = 'branch' | 'commit' | 'tag';
+
+export type DocsVersion = BaseDocsVersion & ExactlyOneKeyWithType<DocsVersionReferenceKind, string>;
+
+/** Latest version must ALWAYS be first */
 export const docsVersions: DocsVersion[] = [
   {
     label: "Version Test 1",
     id: "8.0",
+    number: 8,
```

Comment:
What is the difference between `number` and `id`? Why do you need the two?

Reply:
`number` is a number, used for calculations (e.g. `if (version < 8) { ... }`). `id` is a string, used for routes. I left comments in the type definition.